sqlglot/dialects/tsql.py

```diff
@@ -403,6 +403,11 @@ class Parser(parser.Parser):
             TokenType.END: lambda self: self._parse_command(),
         }
 
+        CONSTRAINT_PARSERS = {
+            **parser.Parser.CONSTRAINT_PARSERS,
+            "PERIOD": lambda self: self._parse_period_for_system_time(),
+        }
```

Comment:
Since this doesn't clash with other constraints, we can move it to the base parser. The philosophy is that whenever we can, we add logic to the base classes so that the dialect they define (or "SQLGlot" dialect) is broader.